### ISSUE TITLE
/cobotta/denso_cobotta_control: open(/dev/denso_cobotta): No such file or directory

### ISSUE THREAD

**Issue by EulerWong** [DENSORobot/denso_cobotta_ros#12]
I connect the cobotta to the switch via ethnet cable, and my computer connect to the switch via ethnet cable too. when i run 

roslaunch denso_cobotta_bringup denso_cobotta_bringup.launch

do you have any tutorial to connect the robot to my pc?

this error occur (shown as in the image below).

OS: ubuntu 18.04
ROS: melodic

![image](https://user-images.githubusercontent.com/6688986/84988114-fbad1080-b173-11ea-8fa3-319b1791178f.png)

**Comment by oishimilk** [comment]
Hi. It seems that the device driver of COBOTTA (`/dev/denso_cobotta`) is not loaded.
`$ sudo modprobe denso_cobotta io_power_mode=1` may be a solution.

I'm not an expert nor employee of denso, information may be incorrect. But I hope it helps you.

**Comment by ShotaAk** [comment]
> do you have any tutorial to connect the robot to my pc?

@EulerWong 
The tutorial page is here (https://densorobot.github.io/docs/denso_cobotta_ros/).
Please refer to the following getting started page that shows requirements to control COBOTTA from a PC installed in the COBOTTA.
https://densorobot.github.io/docs/denso_cobotta_ros/getting_started.html

**Comment by EulerWong** [comment]
> Hi. It seems that the device driver of COBOTTA (`/dev/denso_cobotta`) is not loaded.
> `$ sudo modprobe denso_cobotta io_power_mode=1` may be a solution.
> 
> I'm not an expert nor employee of denso, information may be incorrect. But I hope it helps you.

thanks for you reply and help. it seems that the driver provided by denso just be supported in Ubuntu 16.04, and my Ubuntu version is 18.04. and i still have tried to install the driver,. but fail when run.

when execute the command 
" sudo modprobe  denso_cobotta io_power_mode=1 "
to activate the driver, the error occur:
modprobe: ERROR: could not insert 'denso_cobotta': Exec format error

![image](https://user-images.githubusercontent.com/6688986/85084649-074a1700-b208-11ea-927a-51196d89172e.png)


@DensoWaveRobot can i ask, will you release a driver that can be supported in Ubuntu 18.04. because all of our system and application are based on Ubuntu, and we are not supposed to be downgrade to Ubuntu 16.04 from Ubuntu 18.04. thanks.

**Comment by EulerWong** [comment]
> > do you have any tutorial to connect the robot to my pc?
> 
> @EulerWong
> The tutorial page is here (https://densorobot.github.io/docs/denso_cobotta_ros/).
> Please refer to the following getting started page that shows requirements to control COBOTTA from a PC installed in the COBOTTA.
> https://densorobot.github.io/docs/denso_cobotta_ros/getting_started.html

Hi, ShotaAk, i tried these turorial links, and i realize that  denso cobotta driver only can be supported in Ubuntu 16.04, not Ubuntu 18.04.  as show in this picture. i even chang kinetic to melodic when i run every step in these tutorial, but still fail: /cobotta/denso_cobotta_control: open(/dev/denso_cobotta): No such file or directory

maybe the driver proviced are not supposed to be supported in Ubuntu 18.04, only suppored in Ubuntu 16.04.

![image](https://user-images.githubusercontent.com/6688986/85085143-7ecc7600-b209-11ea-8f06-17a830b68e08.png)

because all of our OS are Ubuntu 18.04, and all of our applications are based on Ubuntu 18.04, even all of our robot are denso cobotta.we donn't want to  downgrade from Ubuntu 18.04 to Ubuntu 16.04.

**Comment by ShotaAk** [comment]
> we donn't want to downgrade from Ubuntu 18.04 to Ubuntu 16.04.

I agree with your comment. But I also have no idea how to install the driver in Ubuntu 18.04.
The version of a COBOTTA in our company (not Denso) is still Ubuntu 16.04 & ROS Kinetic due to run this package.

**Comment by ShotaAk** [comment]
Now, I can suggest only the following temporary solution.

COBOTTA: Installed Ubuntu 16 & Kinetic
Other PC: Installed Ubuntu 18 & Melodic

**Comment by EulerWong** [comment]
> Now, I can suggest only the following temporary solution.
> 
> COBOTTA: Installed Ubuntu 16 & Kinetic
> Other PC: Installed Ubuntu 18 & Melodic

thanks for you help, still not working. i am contacting to denso post-sell support.
